### PR TITLE
Adding maximum entropy method to Frequency Domain Analysis GUI

### LINKED ISSUE #20233: Adding maxEnt to frequency domain GUI
Allow the user to chasnge from FFT to maxEnt algorithm for producing frequency domain plots in the frequency domain analysis GUI.

### PR DESCRIPTION
**Pull request by AnthonyLim23**
The MaxEnt algorithm has been added to the new Muon GUI (Frequency Domain Analysis, see #20205). This involved adding a menu bar that can be used to select the method (FFT or MaxEnt) along with an implementation of the MaxEnt GUI. These changes required some minor restructuring of the code. 

**To test:**
0. Open muon analysis (custom interface) and load some data (the current run will do). Go to the settings tab and bin the data (10) 

1. Open the Frequency Domain analysis GUI. 
2. Use the drop down menu to switch between the two interfaces. 
3. In the MaxEnt interface the data selection should automatically be loaded in the workspace dropdown menu.
4. Try various settings and run MaxEnt. The MaxEnt algorithm is expensive so if you use the Raw data (tick box) then it will take a long time.

Fixes #20233 
**Release Notes** 

The documentation will be updated in #20312
---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by AnthonyLim23** [comment]
This is important to stay in this release

**Comment by mantid-roman** [comment]
Cannot run Frequency Domain Analysis:
```
KeyError: "'MuonAnalysis' does not exist."
  at line 35 in '<Interface>'
  caused by line 20 in '<Interface>'
  caused by line 13 in 'D:/Work/mantid-1/scripts\Muon\transform_presenter.py'
  caused by line 11 in 'D:/Work/mantid-1/scripts\Muon\load_utils.py'
```

**Comment by AnthonyLim23** [comment]
Ah sorry, I forgot to update the instructions. You need to load some data into muon analysis first. It then only considers that data.

**Comment by mantid-roman** [comment]
Could you add a warning to the user with instructions how to make it work if the workspace doesn't exist?

**Comment by mantid-roman** [comment]
What is the output of MaxEnt? I couldn't see any.

**Comment by AnthonyLim23** [comment]
The output is under the group workspace for that run e.g. MUSR00023456 and will have a _maxEnt on the end of the name. 

I will update the documentation later today

**Comment by mantid-roman** [comment]
I merged master. Now I have this error when I start Muon Analysis:
```
Traceback (most recent call last):
  File "D:\Work\mantid\build\bin\Release\pymantidplot\proxies.py", line 73, in _do_dispatch
    raise self.__exception # Ensures this happens in the right thread
TypeError: MantidUI.plot1D(list-of-QString, list-of-int, bool, MantidQt.DistributionFlag distr=MantidQt.DistributionDefault, bool errs=True, GraphOptions.CurveType style=GraphOptions.Unspecified, Graph plotWindow=None, bool clearWindow=False, bool waterfallPlot=False): argument 1 has unexpected type 'list'
TypeError: MantidUI.plot1D(list-of-QString, list-of-int, bool, MantidQt.DistributionFlag distr=MantidQt.DistributionDefault, bool errs=True, GraphOptions.CurveType style=GraphOptions.Unspecified, Graph plotWindow=None, bool clearWindow=False, bool waterfallPlot=False): argument 1 has unexpected type 'list'
  at line 60 in '<Interface>'
  caused by line 30 in '<Interface>'
  caused by line 274 in 'D:\Work\mantid\build\bin\Release\pymantidplot\__init__.py'
  caused by line 99 in 'D:\Work\mantid\build\bin\Release\pymantidplot\proxies.py'
  caused by line 51 in 'D:\Work\mantid\build\bin\Release\pymantidplot\proxies.py'
  caused by line 73 in 'D:\Work\mantid\build\bin\Release\pymantidplot\proxies.py'
```

**Comment by AnthonyLim23** [comment]
I had that this morning,. The solution is to delete everything in the build directory and rebuild it.

**Comment by mantid-roman** [comment]
OK, I'll try.

File: scripts/Muon/transform_selection_presenter.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from __future__ import (absolute_import, division, print_function)
+
+
+class transformSelectionPresenter(object):
```
Comment (mantid-roman) [inline scripts/Muon/transform_selection_presenter.py]: Name should begin with capital T

File: scripts/Muon/transform_view.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+from __future__ import (absolute_import, division, print_function)
+
+from PyQt4 import QtGui
+
+
+class transformView(QtGui.QWidget):
```
Comment (mantid-roman) [inline scripts/Muon/transform_view.py]: Name should begin with capital T

File: scripts/Muon/view_constructor.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+"""
+ simple class to create a single object
+ containing all of the views.
+ Only need to pass a single object to all
+ presenters
+"""
+
+
+class viewConstructor(object):
```
Comment (mantid-roman) [inline scripts/Muon/view_constructor.py]: Class name should start with capital V.

File: scripts/Muon/view_constructor.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+from Muon import FFT_view
+from Muon import MaxEnt_view
+from Muon import transform_selection_view
+
+"""
+ simple class to create a single object
+ containing all of the views.
+ Only need to pass a single object to all
+ presenters
+"""
```
Comment (mantid-roman) [inline scripts/Muon/view_constructor.py]: The docstring should be moved to the line below `class ViewConstructor(object):`

File: scripts/test/Muon/MaxEntPresenter_test.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+class MaxEntPresenterTest(unittest.TestCase):
+    def setUp(self):
+        self.load=mock.create_autospec(load_utils.LoadUtils,spec_set=True)
+        self.load.getCurrentWS=mock.Mock(return_value=["TEST00000001",["fwd","bkwd"]])
+        self.alg=mock.create_autospec(MaxEnt_model.MaxEntThread,spec_set=True)
+        self.alg.start=mock.Mock()
+        self.alg.started=mock.Mock()
+        self.alg.finished=mock.Mock()
+ #       self.alg.started.connect=mock.Mock()
```
Comment (mantid-roman) [inline scripts/test/Muon/MaxEntPresenter_test.py]: Delete the line?

**Review by mantid-roman — CHANGES_REQUESTED** [review]

File: scripts/Muon/load_utils.py [hunk 6]
```diff
@@ -0,0 +1,15 @@
+from __future__ import (absolute_import, division, print_function)
+import mantid.simpleapi as mantid
+
+
+class LoadUtils(object):
+
+    def __init__(self, parent=None):
+        self.options = mantid.AnalysisDataService.getObjectNames()
```
[the remark this answers is not in the document]
Comment (AnthonyLim23) [reply scripts/Muon/load_utils.py]: A few lines done it throws out anything that is not relevant. I will add comments

**Review by mantid-roman — APPROVED** [review]

**Pull request merged**